frontend/cypress/integration/common/overview.ts

```diff
@@ -0,0 +1,135 @@
+    cy.get('button[data-display-mode="' + view + '"]').click();
+});
+
+And(`user filters {string} namespace`, (ns) => {
+    cy.get('select[aria-label="filter_select_type"]')
+        .select('Namespace')
+        .should('have.value', 'namespace_search');
+    cy.get('input[aria-label="filter_input_value"]')
+        .should('exist')
+        .type(ns)
+        .type('{enter}');
+});
+
+And(`user filters {string} health`, (health) => {
+    cy.get('select[aria-label="filter_select_type"]')
+        .select('Health')
+        .should('have.value', 'health');
+    cy.get('select[aria-label="filter_select_value"]')
+        .select(health);
+});
+
+And(`user selects Health for {string}`, (type) => {
+    let innerId = '';
+    switch (type) {
+        case 'Apps':
+            innerId = 'app';
+            break;
+        case 'Workloads':
+            innerId = 'workload';
+            break;
+        case 'Services':
+            innerId = 'service';
+            break;
+    }
+    cy.get('button[aria-labelledby^="overview-type"]')
+        .should('exist')
+        .click()
+        .get('button[id^="' + innerId + '"]')
+        .should('exist')
+        .click();
+});
+
+And(`user sorts by name desc`, () => {
+    cy.get('button[data-sort-asc="true"]')
+        .should('exist')
+        .click();
+});
+
+And(`user selects {string} time range`, (interval) => {
+    let innerId = '';
+    switch (interval) {
+        case 'Last 10m':
+            innerId = '600';
+            break;
+    }
+    cy.get('button[aria-labelledby^="time_range_duration"]')
+        .should('exist')
+        .click()
+        .get('button[id^="' + innerId + '"]')
+        .should('exist')
+        .click();
+});
+
+Then(`user sees the {string} namespace`, (ns) => {
+    cy.get('article[data-namespace="' + ns + '"]').should('exist');
+});
+
+Then(`user doesn't see the {string} namespace`, (ns) => {
+    cy.get('article[data-namespace="' + ns + '"]').should('not.exist');
+});
+
+Then(`user sees a {string} {string} namespace`, (view, ns) => {
+    if (view === "LIST") {
+        cy.get('td[role="gridcell"]').contains(ns).should('exist');
+    } else {
+        cy.get('article[data-namespace="' + ns + '"][data-display-mode="' + view + '"]').should('exist');
+    }
+});
+
+Then(`user sees the {string} namespace with {string}`, (ns, type) => {
+    let innerType = '';
+    switch (type) {
+        case 'Applications':
+            innerType = 'app';
+            break;
+        case 'Workloads':
+            innerType = 'workload';
+            break;
+        case 'Services':
+            innerType = 'service';
+            break;
+    }
+    cy.get('article[data-namespace="' + ns + '"]')
+        .should('exist')
+        .find('[data-overview-type="' + innerType + '"]')
+        .should('exist');
+});
+
+Then(`user sees the {string} namespace list`, (nslist) => {
+    const nss = nslist.split(',');
+    cy.get('article')
+        .should('have.length', 2)
```

Comment:
Is this testing the number of namespaces in the dropdown? imo we don't want hardcoded numbers in the tests for things like this since they will change over time and they might also change depending on whether we are testing in upstream or downstream. If anything, we should get the number of namespaces we expect with something like `cy.exec('kubectl get ns')` and compare that to the number of namespaces in the list.

Reply:
oh, yes, good point, it should compare with the length of the nslist.
But I'm -1 to call the system for a `kubectl get ns` unless there is a good reason.

I'll update the code.

Reply:
Even better!